hasktorch/src/Torch/NN.hs

```diff
@@ -107,9 +113,17 @@ data LinearSpec = LinearSpec { in_features :: Int, out_features :: Int }
 
 data Linear = Linear { weight :: Parameter, bias :: Parameter } deriving (Show, Generic)
 
+linear' :: Linear -> Tensor -> Tensor
```

Comment:
Why the `'`?

Reply:
This is a holdover from initial exploration when a few existing `linear` implementations were in the examples and I was trying to avoid breaking them at first. 

However, now that all untyped implementations are standardized to the library implementation and use the pytorch dimension conventions (they had to be in order to share the Randomizable typeclass), it makes since to rename this and its occurrences in examples to `linear`.

Reply:
I've replaced `linear'` with `linear`, leaving the name clash resolution up to the user to determine at import time for now.

Going forward we should try to land on a convention for the function version vs. the version that operates on `Parameter` state.